net.c

```diff
@@ -375,6 +399,15 @@ static int _redisContextConnectTcp(redisContext *c, const char *addr, int port,
                 goto error;
             }
         }
+
+        /* For repeat connection */
+        if (c->saddr) {
+            free(c->saddr);
+        }
+        c->saddr = malloc(sizeof(*p->ai_addr));
```

Comment:
The allocation should be done based on `ai_addrlen`, as `ai_addr` is a vanilla sockaddr which may be smaller than what you really need (e.g. for AF_UNIX or AF_INET6).

Reply:
oops - done. There's also `struct sockaddr_storage`, but ideally I'd like to preserve the original `addrinfo` structure anyway.